nixos/modules/programs/fish.nix

```diff
@@ -53,7 +55,24 @@ in
         type = lib.types.bool;
       };
 
-      package = lib.mkPackageOption pkgs "fish" { };
+      package = lib.mkOption {
+        type = lib.types.package;
+        default = pkgs.fishWrapper;
+        defaultText = lib.literalExpression "pkgs.fish";
+        example = lib.literalExpression ''
+          pkgs.fishWrapper.override {
+            pluginPkgs = with pkgs.fishPlugins; [
+              autopair
+              fifc
+              done
+            ];
+          }
+        '';
```

Comment:
How would users be suggested to use fish, `programs.fish.plugins` or `programs.fish.package = pkgs.fishWrapper.override` ?

Do not include both implementations of the same feature in your user documentation, especially if one of them is a legacy implementation.